sdk/dotnet/Pulumi.Automation/ProjectTemplateConfigValue.cs

```diff
@@ -12,5 +15,34 @@ public class ProjectTemplateConfigValue
         public string? Default { get; set; }
 
         public bool? Secret { get; set; }
+
+        internal static IEqualityComparer<ProjectTemplateConfigValue> Comparer { get; } = new ProjectTemplateConfigValueComparer();
+
+        private sealed class ProjectTemplateConfigValueComparer : IEqualityComparer<ProjectTemplateConfigValue>
+        {
+            bool IEqualityComparer<ProjectTemplateConfigValue>.Equals(ProjectTemplateConfigValue? x, ProjectTemplateConfigValue? y)
+            {
+                if (x == null)
+                {
+                    return y == null;
+                }
+
+                if (y == null)
+                {
+                    return x == null;
+                }
+
+                return x.Description == y.Description && x.Default == y.Default && x.Secret == y.Secret;
+            }
+
+            int IEqualityComparer<ProjectTemplateConfigValue>.GetHashCode(ProjectTemplateConfigValue obj)
+            {
+                var hash = new HashCode();
+                hash.Add(obj.Description);
+                hash.Add(obj.Default);
+                hash.Add(obj.Secret);
+                return hash.ToHashCode();
```

Comment:
Can consider using [HashCode.Combine](https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=net-5.0)

```suggestion
                return HashCode.Combine(obj.Description, obj.Default, obj.Secret);
```